### PR TITLE
Apply various hlint hints

### PR DESCRIPTION
**Pull request by markus1189**
Appy various hlint hints:
- Attoparsec.Args: Replace not . (==) with (/=)
- Stack.Build: redundant brackets
- Stack.Build.Installed: use `forM_`, redundant bracket
- Stack.BuildPlan: fuse mapM and map, use const
- Stack.Build.Source: redundant bracket, eta reduce
- Stack.Config: Redundant brackets
- Stack.Config.Nix: use when
- Stack.Coverage: use (/=)
- Stack.Docker.GlobalDB: use unless
- Stack.Docker: redundant brackets
- Stack.Ghci:  use mapMaybe
- Stack.Image: reduce duplication

-- 24pullrequests
[](http://24pullrequests.com/)

### CODE REVIEW

File: src/Stack/BuildPlan.hs [hunk 1]
```diff
@@ -638,7 +637,7 @@ shadowMiniBuildPlan :: MiniBuildPlan
 shadowMiniBuildPlan (MiniBuildPlan cv pkgs0) shadowed =
     (MiniBuildPlan cv $ Map.fromList met, Map.fromList unmet)
   where
-    pkgs1 = Map.difference pkgs0 $ Map.fromSet (\_ -> ()) shadowed
+    pkgs1 = Map.difference pkgs0 $ Map.fromSet const shadowed
```
Comment (mgsloan) [inline src/Stack/BuildPlan.hs]: Can you please leave this one out?  This actually causes the type of this map to change.  `(\_ -> ())` isn't the same as `const`.  Personally, I don't agree with some of hlint's suggestions.  Using `const` is one thing I don't like.  Compare:

`const ()`
`\_ -> ()`

The lambda version is shorter and more straightforward and naturally scales:

`const (const ())`
`\_ _ -> ()`

File: src/Stack/Docker/GlobalDB.hs [hunk 2]
```diff
@@ -76,11 +76,11 @@ getDockerImagesLastUsed config =
...
-  where go = do l <- selectList [] []
-                forM_ l (\(Entity k (DockerImageProject{dockerImageProjectImageHash = h})) ->
-                           if h `elem` existingHashes
-                             then return ()
-                             else delete k)
+  where
+    go = do
+      l <- selectList [] []
+      forM_ l (\(Entity k DockerImageProject{dockerImageProjectImageHash = h}) ->
+                   unless (h `elem` existingHashes) $ delete k)
```
Comment (mgsloan) [inline src/Stack/Docker/GlobalDB.hs]: How about `when (h `notElem` existingHashes)`?

I find `when` much clearer than `unless` in general, it reads way better.

**Comment by mgsloan** [comment]
Thanks!  I'll merge when my comments are addressed

**Comment by markus1189** [comment]
Thanks for the thorough, feedback, agreed in all points!

**Pull request merged**